libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -31,6 +37,83 @@ void AP_DDS_Client::update_topic(builtin_interfaces_msg_Time& msg)
 
 }
 
+void AP_DDS_Client::update_topic(sensor_msgs_msg_NavSatFix& msg)
+{
+    // Add a lambda that takes in navsatfix msg and populates the cov
+    // Make it constexpr if possible
+    // https://www.fluentcpp.com/2021/12/13/the-evolutions-of-lambdas-in-c14-c17-and-c20/
+    // constexpr auto times2 = [] (sensor_msgs_msg_NavSatFix* msg) { return n * 2; };
+
+
+    update_topic(msg.header.stamp);
+    strcpy(msg.header.frame_id, WGS_84_FRAME_ID);
+    msg.status.service = 0; // SERVICE_GPS
+    msg.status.status = -1; // STATUS_NO_FIX
+
```

Comment:
we should do:
```
 auto &gps = AP::gps();
 WITH_SEMAPHORE(gps.get_semaphore());
```
and use gps instead of AP::gps() in the rest of the fn
taking the semaphore will guarantee we get consistent data from one point in time

Reply:
ff9e694022612108780c0887b4d6d945c4eb1b03